### PR TITLE
fix translation and add feature "Bigger icons in calendar view (optional)"

### PR DESCRIPTION
**Pull request by mroswald**

### CODE REVIEW

**Comment by mroswald** [comment]
please provide information about used js combine / minify tool, want to use the exact same. how do you do the translations?

**Comment by arnecls** [comment]
Did not have the chance to look through the mergerequest (and probably won't have until tuesday).

So just a few quick answers on your questions:
- I use google closure compiler with simple optimizations on the output genarated by lib/scirpt/allscripts.php
- Translations are community driven (except for german and english) so I normally use NULL for new translations and ask for proper translations. If there are no translations before a release I try my best with google translate / some interpretation of what's already been translated.

**Comment by mroswald** [comment]
are you comfortable with my process using yuicompressor? I would remove if it you're not

**Comment by arnecls** [comment]
As long as the file is small and contains no errors the method does not matter ;)
As long as the raidplaner scipts do not use proper namespaces the more agressive methods form closure compiler don't work anyway so I guess yuicompressor is ok here.

File: lib/private/message_settings_update.php [hunk 1]
```diff
@@ -70,9 +70,29 @@ function generateQueryStringText( $CurrentValues, &$BindValues, $ValueName, $New
     
         return '';
     }
-    
+
     // -----------------------------------------------------------------------------
-    
+
+    function getRequestValue( $aRequest, $sParameterName, $sValueType = "IntValue" )
```
Comment (arnecls) [inline lib/private/message_settings_update.php]: Good idea, but I would change two things:
1. Either use constants for sValueType or better still two different functions (one per type) to get rid of the if/switch altogether
2. Pass the value to change as a reference to this function. This way you can make sure that intvalues ar not assigned textvalues + you could even clear the other value.

File: lib/script/allscripts.php [hunk 2]
```diff
@@ -38,6 +40,53 @@
             echo "<script type=\"text/javascript\" src=\"lib/script/".$File."?v=".$gVersion."\"></script>\n";
         }
     }
+    elseif (PHP_SAPI == "cli")
```
Comment (arnecls) [inline lib/script/allscripts.php]: Ah - now I see what you meant.
Please no tooling in this area. This should be moved to an external script, e.g. Grunt file or sth. like that.

**Comment by arnecls** [comment]
I still need to check the visuals. Hopefully I get to do that tomorrow.

File: lib/private/message_settings_update.php [hunk 3]
```diff
@@ -70,9 +70,29 @@ function generateQueryStringText( $CurrentValues, &$BindValues, $ValueName, $New
...
+    {
+        if (is_array($aRequest) && array_key_exists($sParameterName, $aRequest))
+        {
+            return $sParameterName;
```
Comment (arnecls) [inline lib/private/message_settings_update.php]: You mean $aRequest[$sParameterName]

Comment (arnecls) [inline lib/private/message_settings_update.php]: ...and while we're at it - if there already is a function in place checking the request intval and strval should be involved here. Another good point for having two functions btw.

File: lib/script/allscripts.php (hunk 2 above)
Comment (mroswald) [inline lib/script/allscripts.php]: ok, reverted

File: lib/private/message_settings_update.php (hunk 3 above)
Comment (mroswald) [inline lib/private/message_settings_update.php]: Sure, $aRequest[$sParameterName]. Sorry for that

You mean just 
$Settings['PurgeRaids']['IntValue']      = intval($aRequest['purgeTime']);
?? That wouldn't catch the PHP notice if the index "purgeTime" is missing in $aRequest.

File: lib/layout/raidlist.css [hunk 4]
```diff
@@ -71,6 +71,35 @@
...
+#raidlist .raidSlot .overlayStatus .ui-icon {
+    display: inline-block;
+}
+
+#raidlist .raidSlot .overlayStatus:before {
+    width: 28px;
+    height: 28px;
+    display: inline-block;
+    content: '';
+    margin-right: 16px;
```
Comment (arnecls) [inline lib/layout/raidlist.css]: This is probably not a good idea as the position of the new element depends on the size of the element to it's right which is - afair - a number that can be either 1 or 2 digits wide.
In addition to that I would use a separate element, not before. This way you will see in the script that there is an element and don't have to look it up in CSS first.

File: lib/private/message_settings_update.php (hunk 3 above)
Comment (arnecls) [inline lib/private/message_settings_update.php]: function getRequestIntValue($aRequest, $sParameterName, &$aValue)
{
    $aValue = (is_array($aRequest) && array_key_exists($sParameterName, $aRequest))
        ? intval($aRequest[$sParameterName]
        : 0
}

Another function like that for strings, too.

Comment (mroswald) [inline lib/private/message_settings_update.php]: which still is a if statement :)

-> changed

File: lib/layout/raidlist.css (hunk 4 above)
Comment (mroswald) [inline lib/layout/raidlist.css]: please see the newer commits, changed to a not-pseudo html element as you suggested and fixed the issue to the size depency

**Comment by arnecls** [comment]
Functionality checked and working :+1: 
The visuals need some improvement though:

big_icons01
big_icons02

1. The lock icon is misplaced
2. The raid size could be displayed in a larger font as there is now more space
3. When attended but not set up the role size is misplaced. See additional notes on this below
4. The raidsize has moved up for locked/unattended benched statuses
5. The "ok" sign is not visible on certain icons and should be removed as it contains no additional information to the "traffic light" status of the raid size. See additional notes on this below

Remarks (on 3 and 5):
I would remove the role information on attended-but-not-setup (aka benched) status raids because this way the display of the role icon becomes a clear indicator for "you are now set up as ...".
As the role icon is a far bigger eye catcher than the number one might get false impressions by just looking at the icon. 

I would also remove any additional information (the ok sign) from the small icons as they are already packed with two overlays and some guild use the lower left corner to display information about the dungeon mode.
I do know that the traffic light is not the best indicator (as you can see quite well on the two images) but given the small amount of space this was the best I could come up with.
A new indicator is probably worth a whole new discussion and/or pull request.

The rest of the merge request looks perfectly fine for me now.

**Comment by mroswald** [comment]
Thanks for your testing and feedback. I will fix this in the next 2 days. Using the color of the raidsize as indicator was not really clear as for color-blind/-weak people it's not visible. I agree that the small visual should not be touched by my changes.

Using the role icon as setup-indicator was my first idea, but thinking/developing a bit more of this I thought it was a good idea to see the role AND the status without using the tooltip.

I will follow your advice to use the role icon as setup indicator, but maybe we'll find a smart solution for displaying more information since the icon as itself isn't that important. I'm not used to many other games, but for wow only one or maybe to raid instances are visited at the same time.

**Comment by mroswald** [comment]
fixed your points mentioned in the screenshot (style bugs)

**Comment by arnecls** [comment]
I will re-test it during the next few days.
Thanks for the update.

**Comment by arnecls** [comment]
Merged into a new feature branch and fixed the last few layout issues regarding the element positions.
Good work.

Note: As I wrote some time ago on google+, all new features will go into branches, only fixes go into the trunk. If a new version is released all feature branches will be merged into the trunk before release.
As of this the feature is now counting as accepted and the merge request is closed.

**Pull request closed without merging**